go/context_headers.go

```diff
@@ -0,0 +1,17 @@
+package main
+
+import (
+	"context"
+	"net/http"
+)
+
+func extractHeaders(ctx context.Context, r *http.Request) context.Context {
+	header := r.Header.Get("X-Telepresence-Id")
```

Comment:
Is there a reason for the different case here (vs what we're propagating?)

Reply:
It's what was coming through in logs, I can check if all lower case works, not sure quite how r.Header.Get does on case matching.